server/workflow/workflow_server.go

```diff
@@ -140,6 +150,23 @@ func (s *workflowServer) ListWorkflows(ctx context.Context, req *workflowpkg.Wor
 	if err != nil {
 		return nil, sutils.ToStatusError(err, codes.Internal)
 	}
+	archivedWfList, err := s.wfArchiveServer.ListArchivedWorkflows(ctx, &workflowarchivepkg.ListArchivedWorkflowsRequest{
+		ListOptions: listOption,
+		NamePrefix:  "",
+		Namespace:   req.Namespace,
+	})
+	if err != nil {
+		log.Warnf("unable to list archived workflows:%v", err)
+	} else {
+		if archivedWfList != nil {
+			for _, item := range archivedWfList.Items {
+				if !containsWorkflow(*wfList, item) {
+					wfList.Items = append(wfList.Items, item)
+				}
```

Comment:
I see a O^2 problem here. We iterate `wfList` every time for `archivedWfList`. I think we need to improve this by building up a map of UIDs that we seen in the live.

Reply:
Thanks for the feedback! I've improved this in the changes.